TST: mpl-dev job, test_misc.py failing

UPDATE:
- [ ] When upstream issue is resolved, undo #11292

Example log: https://app.circleci.com/pipelines/github/astropy/astropy/3915/workflows/c3d3cbd1-dfa9-44a0-9cf6-efa363ef032c/jobs/84216

```
platform linux -- Python 3.7.3, pytest-6.2.1, py-1.10.0, pluggy-0.13.1
cachedir: .tox/py37-test-image-mpldev/.pytest_cache
Matplotlib: 3.3.2+2191.gb52aad8
Freetype: 2.6.1

Running tests with Astropy version 4.3.dev531+gfc941c3.
Running tests in astropy docs.

Date: 2021-01-22T11:27:05

Platform: Linux-4.15.0-1077-aws-x86_64-with-debian-stretch-sid

Executable: /root/project/.tox/py37-test-image-mpldev/bin/python

Full Python Version: 
3.7.3 (default, Jan  9 2020, 12:06:30) 
[GCC 5.4.0 20160609]

encodings: sys: utf-8, locale: UTF-8, filesystem: utf-8
byteorder: little
float info: dig: 15, mant_dig: 15

Package versions: 
Numpy: 1.19.5
Scipy: not available
Matplotlib: 3.3.2+2191.gb52aad8
h5py: not available
Pandas: not available
PyERFA: 1.7.1.1
Cython: not available
Scikit-image: not available
asdf: not available

Using Astropy options: remote_data: astropy.

...

__________________________ test_invalid_slices_errors __________________________

ignore_matplotlibrc = None

    def test_invalid_slices_errors(ignore_matplotlibrc):
    
        # Make sure that users get a clear message when specifying a WCS with
        # >2 dimensions without giving the 'slices' argument, or if the 'slices'
        # argument has too many/few elements.
    
        wcs3d = WCS(naxis=3)
        wcs3d.wcs.ctype = ['x', 'y', 'z']
    
        plt.subplot(1, 1, 1, projection=wcs3d, slices=('x', 'y', 1))
    
        with pytest.raises(ValueError) as exc:
>           plt.subplot(1, 1, 1, projection=wcs3d)
E           Failed: DID NOT RAISE <class 'ValueError'>

../../.tox/py37-test-image-mpldev/lib/python3.7/site-packages/astropy/visualization/wcsaxes/tests/test_misc.py:345: Failed
__________________________________ test_repr ___________________________________

ignore_matplotlibrc = None

    def test_repr(ignore_matplotlibrc):
    
        # Unit test to make sure __repr__ looks as expected
    
        wcs3d = WCS(GAL_HEADER)
    
        # Cube header has world coordinates as distance, lon, lat, so start off
        # by slicing in a way that we select just lon,lat:
    
        ax = plt.subplot(1, 1, 1, projection=wcs3d, slices=(1, 'x', 'y'))
        assert repr(ax.coords) == EXPECTED_REPR_1
    
        # Now slice in a way that all world coordinates are still present:
    
        ax = plt.subplot(1, 1, 1, projection=wcs3d, slices=('x', 'y', 1))
>       assert repr(ax.coords) == EXPECTED_REPR_2
E       AssertionError: assert '<Coordinates...     yes\n\n>' == '<Coordinates...     yes\n\n>'
E         Skipping 266 identical leading characters in diff, use -v to show
E         Skipping 152 identical trailing characters in diff, use -v to show
E         -           yes
E         ?           ^^^
E         +            no
E         ?           ^^^
E                 1

../../.tox/py37-test-image-mpldev/lib/python3.7/site-packages/astropy/visualization/wcsaxes/tests/test_misc.py:418: AssertionError
=========================== short test summary info ============================
FAILED ../../.tox/py37-test-image-mpldev/lib/python3.7/site-packages/astropy/visualization/wcsaxes/tests/test_misc.py::test_invalid_slices_errors
FAILED ../../.tox/py37-test-image-mpldev/lib/python3.7/site-packages/astropy/visualization/wcsaxes/tests/test_misc.py::test_repr
```

cc @astrofrog and @Cadair

--- a/astropy/visualization/wcsaxes/tests/test_misc.py
+++ b/astropy/visualization/wcsaxes/tests/test_misc.py
@@ -1,5 +1,7 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 
+from contextlib import nullcontext
+
 from packaging.version import Version
 import pytest
 import numpy as np
@@ -23,6 +25,10 @@ from astropy.visualization.wcsaxes.transforms import CurvedTransform
 ft_version = Version(matplotlib.ft2font.__freetype_version__)
 FREETYPE_261 = ft_version == Version("2.6.1")
 TEX_UNAVAILABLE = not matplotlib.checkdep_usetex(True)
+
+# TODO: Improve check when this issue is resolved:
+# https://github.com/matplotlib/matplotlib/issues/19419
+MATPLOTLIB_DEV = '+' in matplotlib.__version__
 
 
 def teardown_function(function):
@@ -341,15 +347,22 @@ def test_invalid_slices_errors(ignore_matplotlibrc):
 
     plt.subplot(1, 1, 1, projection=wcs3d, slices=('x', 'y', 1))
 
-    with pytest.raises(ValueError) as exc:
-        plt.subplot(1, 1, 1, projection=wcs3d)
-    assert exc.value.args[0] == ("WCS has more than 2 pixel dimensions, so "
-                                 "'slices' should be set")
+    if MATPLOTLIB_DEV:
+        ctx = nullcontext()
+    else:
+        ctx = pytest.raises(ValueError)
 
-    with pytest.raises(ValueError) as exc:
+    with ctx as exc:
+        plt.subplot(1, 1, 1, projection=wcs3d)
+    if exc:
+        assert exc.value.args[0] == ("WCS has more than 2 pixel dimensions, so "
+                                     "'slices' should be set")
+
+    with ctx as exc:
         plt.subplot(1, 1, 1, projection=wcs3d, slices=('x', 'y', 1, 2))
-    assert exc.value.args[0] == ("'slices' should have as many elements as "
-                                 "WCS has pixel dimensions (should be 3)")
+    if exc:
+        assert exc.value.args[0] == ("'slices' should have as many elements as "
+                                     "WCS has pixel dimensions (should be 3)")
 
     wcs2d = WCS(naxis=2)
     wcs2d.wcs.ctype = ['x', 'y']
@@ -363,15 +376,21 @@ def test_invalid_slices_errors(ignore_matplotlibrc):
     ax = plt.subplot(1, 1, 1, projection=wcs2d, slices=['x', 'y'])
     assert ax.frame_class is RectangularFrame
     ax = plt.subplot(1, 1, 1, projection=wcs2d, slices=(1, 'x'))
-    assert ax.frame_class is RectangularFrame1D
+    if MATPLOTLIB_DEV:
+        assert ax.frame_class is RectangularFrame
+    else:
+        assert ax.frame_class is RectangularFrame1D
 
     wcs1d = WCS(naxis=1)
     wcs1d.wcs.ctype = ['x']
 
     ax = plt.subplot(1, 1, 1, projection=wcs1d)
-    assert ax.frame_class is RectangularFrame1D
+    if MATPLOTLIB_DEV:
+        assert ax.frame_class is RectangularFrame
+    else:
+        assert ax.frame_class is RectangularFrame1D
 
-    with pytest.raises(ValueError):
+    with ctx:
         plt.subplot(1, 1, 1, projection=wcs2d, slices=(1, 'y'))
 
 
@@ -399,6 +418,18 @@ EXPECTED_REPR_2 = """
 >
  """.strip()
 
+EXPECTED_REPR_2_MPLDEV = """
+<CoordinatesMap with 3 world coordinates:
+
+  index            aliases                type   unit wrap format_unit visible
+  ----- ------------------------------ --------- ---- ---- ----------- -------
+      0                   distmod dist    scalar      None                  no
+      1 pos.galactic.lon glon-car glon longitude  deg  360         deg     yes
+      2 pos.galactic.lat glat-car glat  latitude  deg None         deg     yes
+
+>
+ """.strip()
+
 
 def test_repr(ignore_matplotlibrc):
 
@@ -415,7 +446,10 @@ def test_repr(ignore_matplotlibrc):
     # Now slice in a way that all world coordinates are still present:
 
     ax = plt.subplot(1, 1, 1, projection=wcs3d, slices=('x', 'y', 1))
-    assert repr(ax.coords) == EXPECTED_REPR_2
+    if MATPLOTLIB_DEV:
+        assert repr(ax.coords) == EXPECTED_REPR_2_MPLDEV
+    else:
+        assert repr(ax.coords) == EXPECTED_REPR_2
 
 
 @pytest.fixture

--- a/astropy/visualization/wcsaxes/tests/test_misc.py
+++ b/astropy/visualization/wcsaxes/tests/test_misc.py
@@ -1,7 +1,5 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 
-from contextlib import nullcontext
-
 from packaging.version import Version
 import pytest
 import numpy as np
@@ -336,6 +334,7 @@ def test_iterate_coords(ignore_matplotlibrc, tmpdir):
     x, y, z = ax.coords
 
 
+@pytest.mark.xfail(MATPLOTLIB_DEV, reason="https://github.com/matplotlib/matplotlib/issues/19432")
 def test_invalid_slices_errors(ignore_matplotlibrc):
 
     # Make sure that users get a clear message when specifying a WCS with
@@ -347,22 +346,15 @@ def test_invalid_slices_errors(ignore_matplotlibrc):
 
     plt.subplot(1, 1, 1, projection=wcs3d, slices=('x', 'y', 1))
 
-    if MATPLOTLIB_DEV:
-        ctx = nullcontext()
-    else:
-        ctx = pytest.raises(ValueError)
-
-    with ctx as exc:
+    with pytest.raises(ValueError) as exc:
         plt.subplot(1, 1, 1, projection=wcs3d)
-    if exc:
-        assert exc.value.args[0] == ("WCS has more than 2 pixel dimensions, so "
-                                     "'slices' should be set")
+    assert exc.value.args[0] == ("WCS has more than 2 pixel dimensions, so "
+                                 "'slices' should be set")
 
-    with ctx as exc:
+    with pytest.raises(ValueError) as exc:
         plt.subplot(1, 1, 1, projection=wcs3d, slices=('x', 'y', 1, 2))
-    if exc:
-        assert exc.value.args[0] == ("'slices' should have as many elements as "
-                                     "WCS has pixel dimensions (should be 3)")
+    assert exc.value.args[0] == ("'slices' should have as many elements as "
+                                 "WCS has pixel dimensions (should be 3)")
 
     wcs2d = WCS(naxis=2)
     wcs2d.wcs.ctype = ['x', 'y']
@@ -376,21 +368,15 @@ def test_invalid_slices_errors(ignore_matplotlibrc):
     ax = plt.subplot(1, 1, 1, projection=wcs2d, slices=['x', 'y'])
     assert ax.frame_class is RectangularFrame
     ax = plt.subplot(1, 1, 1, projection=wcs2d, slices=(1, 'x'))
-    if MATPLOTLIB_DEV:
-        assert ax.frame_class is RectangularFrame
-    else:
-        assert ax.frame_class is RectangularFrame1D
+    assert ax.frame_class is RectangularFrame1D
 
     wcs1d = WCS(naxis=1)
     wcs1d.wcs.ctype = ['x']
 
     ax = plt.subplot(1, 1, 1, projection=wcs1d)
-    if MATPLOTLIB_DEV:
-        assert ax.frame_class is RectangularFrame
-    else:
-        assert ax.frame_class is RectangularFrame1D
+    assert ax.frame_class is RectangularFrame1D
 
-    with ctx:
+    with pytest.raises(ValueError):
         plt.subplot(1, 1, 1, projection=wcs2d, slices=(1, 'y'))
 
 
@@ -418,19 +404,8 @@ EXPECTED_REPR_2 = """
 >
  """.strip()
 
-EXPECTED_REPR_2_MPLDEV = """
-<CoordinatesMap with 3 world coordinates:
 
-  index            aliases                type   unit wrap format_unit visible
-  ----- ------------------------------ --------- ---- ---- ----------- -------
-      0                   distmod dist    scalar      None                  no
-      1 pos.galactic.lon glon-car glon longitude  deg  360         deg     yes
-      2 pos.galactic.lat glat-car glat  latitude  deg None         deg     yes
-
->
- """.strip()
-
-
+@pytest.mark.xfail(MATPLOTLIB_DEV, reason="https://github.com/matplotlib/matplotlib/issues/19432")
 def test_repr(ignore_matplotlibrc):
 
     # Unit test to make sure __repr__ looks as expected
@@ -446,10 +421,7 @@ def test_repr(ignore_matplotlibrc):
     # Now slice in a way that all world coordinates are still present:
 
     ax = plt.subplot(1, 1, 1, projection=wcs3d, slices=('x', 'y', 1))
-    if MATPLOTLIB_DEV:
-        assert repr(ax.coords) == EXPECTED_REPR_2_MPLDEV
-    else:
-        assert repr(ax.coords) == EXPECTED_REPR_2
+    assert repr(ax.coords) == EXPECTED_REPR_2
 
 
 @pytest.fixture

